File: Backend/src/main/java/org/example/smartparking/entity/ParkingSlot.java
package org.example.smartparking.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "parking_slots")
@Getter @Setter
@NoArgsConstructor @AllArgsConstructor
@Builder
public class ParkingSlot {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 20)
    private String slotNumber;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    @Builder.Default
    private SlotStatus status = SlotStatus.AVAILABLE;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    @Builder.Default
    private SlotType slotType = SlotType.STANDARD;

    @Column(length = 10)
    private String floor;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "parking_id", nullable = false)
    private Parking parking;

    @OneToMany(mappedBy = "parkingSlot", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<Reservation> reservations = new ArrayList<>();

    @CreationTimestamp
    @Column(updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    private LocalDateTime updatedAt;
}

File: mobile/lib/models/parking_slot.dart
class ParkingSlot {
  final int id;
  final String slotNumber;
  final String status;
  final String slotType;
  final String? floor;
  final int parkingId;
  final String parkingName;
  final DateTime? createdAt;

  ParkingSlot({
    required this.id,
    required this.slotNumber,
    required this.status,
    required this.slotType,
    this.floor,
    required this.parkingId,
    required this.parkingName,
    this.createdAt,
  });

  factory ParkingSlot.fromJson(Map<String, dynamic> json) => ParkingSlot(
        id: json['id'] as int,
        slotNumber: json['slotNumber'] as String,
        status: json['status'] as String,
        slotType: json['slotType'] as String,
        floor: json['floor'] as String?,
        parkingId: json['parkingId'] as int,
        parkingName: json['parkingName'] as String,
        createdAt: json['createdAt'] != null
            ? DateTime.parse(json['createdAt'] as String)
            : null,
      );

  bool get isAvailable => status == 'AVAILABLE';
}
